File: java/data_src_dependent/bst_gutenberg/bst_gutenberg.java

//
// Testing the gutenberg books data with a binary search tree
// 
import java.util.Scanner;
import bridges.connect.Bridges;
import bridges.base.BSTElement;
import bridges.data_src_dependent.GutenbergBook;
import java.util.ArrayList;

public class bst_gutenberg {
	public static final int maxElements = 100; //number of tweets

	public static void main(String[] args) throws Exception{
	
    						// Instantiate a Bridges object 
		Bridges bridges = new Bridges (101, args[0], args[1]);
		bridges.setServer (args[2]);

		bridges.setTitle("GutenBerg Book Collection(Meta Data Only)");


							// Retrieve Gutenberg Book Data 
		ArrayList<GutenbergBook> book_list = 
				(ArrayList<GutenbergBook>) Bridges.getGutenbergBookMetaData();

		BST<String, GutenbergBook>  bst = new BST<String, GutenbergBook>();
		for ( int i = 0; i < book_list.size(); i++ ){
			bst.insert(book_list.get(i).getTitle(), book_list.get(i));
		}

							// give BRIDGES handle to tree root
		bridges.setDataStructure(bst.getTreeRoot());
							// visualize the tree
		bridges.visualize();
	}
}

File: c++/data_src_dependent/bst_gutenberg/bst_gutenberg.cpp
#include <string>

using namespace std;

#include "Bridges.h"
#include "DataSource.h"
#include "data_src/GutenbergBook.h"
#include "BSTElement.h"
#include "BST.h"

#define LOCAL_SERVER 0

using namespace bridges;


int main(int argc, char **argv) {
	string hilite_color = "orange", 
			def_color = "green",
			end_color = "red";
		
#if TESTING
                        // command line args provide credentials and server to test on
    Bridges bridges (atoi(argv[1]), argv[2], argv[3]);
    if (argc > 4)
        bridges.setServer(argv[4]);
#else
    Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", 
                                "YOUR_API_KEY");
#endif

								// read the earth quake  data and build the BST

	bridges.setTitle("BST Example: Gutenberg Book Collection");

	DataSource ds (&bridges);

    vector<GutenbergBook> book_list = ds.getGutenbergBookData();

    BST<string, GutenbergBook> bst;

	for (int k = 0; k < book_list.size(); k++) {
		GutenbergBook b = book_list[k];
		bst.insert(b.getTitle(), b);
	}


					// visualize the binary search tree
	bridges.setDataStructure(bst.getRoot());
	bridges.visualize();

	return 0;
}
